src/concrete/ml/onnx/onnx_impl_utils.py

```diff
@@ -223,3 +224,26 @@ def onnx_avgpool_compute_norm_const(
         norm_const = numpy.prod(kernel_shape)
 
     return norm_const
+
+
+def rounded_comparison(
+    x: numpy.ndarray, y: numpy.ndarray, lsbs_to_remove: int, operation: Callable
+) -> Tuple[bool]:
+    """Comparison operation using AutoRounder.
+
+    Args:
+        x (numpy.ndarray): Input tensor
+        y (numpy.ndarray): Input tensor
+        lsbs_to_remove (int): The number of the least significant bits to remove
+        operation (Callable): Comparison operation
+
+    Returns:
+        Tuple[bool]: If x and y satisfy the comparison operator.
+    """
+
+    assert isinstance(lsbs_to_remove, int)
+
+    half = 1 << (lsbs_to_remove - 1)
+    rounded_subtraction = round_bit_pattern((x - y) - half, lsbs_to_remove=lsbs_to_remove)
+
+    return (operation(rounded_subtraction),)
```

Comment:
this is not exactly a rounded comparison then, as you only return op(x, y) and not the comparison boolean itself) ! Why not directly computing this op over `0` and return the result ?

Reply:
the op must return a boolean. maybe this could be used:

```
import typing

ComparisonOperationType = typing.Callable[[int], bool]

def rounded_comparison(
    x: numpy.ndarray, y: numpy.ndarray, lsbs_to_remove: int, operation: ComparisonOperationType
) -> Tuple[bool]:
```

Reply:
yes it should return a boolean indeed, but since we are comparing to a soustraction here, it should always be compared to 0 right ? 

I mean returning something like `operation(rounded_subtraction, 0)`, with `operation` being `lambda x, t: x >= t` for `GreaterOrEqual` op ! Actually, I've just learned that we could even use the `operator` library (to import), and therefore have `operator.ge` for `GreaterOrEqual` for example

Reply:
it's just that this function is meant to compute a comparison between x and y with rounding, imo I can't see why we could provide anything else that a comparison with 0 then 😅  And it would still be a boolean at the end, the only difference is that `operation` would only need to be a comparison operator, not a lambda 

although from what I understood, `rounded_comparison` is not symmetric between x and y, so I guess the main point of giving a lambda instead of an comparison op on for `operation` is only because we need to make sure that we are giving the right "comparison order" 

so I guess I'm ok to keep it that way, I would maybe at least explain a bit why it's done like this (and more importantly, warn that this function is not symmetric !)